From c2aabe13bcf6c9f67f8bce3496dfd0cffb4d794e Sun Apr  2 00:00:00 2001
From: =?UTF-8?q?B=C5=82a=C5=BCej=20Szczygie=C5=82?= <spaz16@wp.pl>
Date: Sun,  2 Apr 2023 21:38:47 +0200
Subject: [PATCH] Apply always on top to settings window
         [PATCH] Apply always on top to about window

Amends: c2aabe13bcf6c9f67f8bce3496dfd0cffb4d794e
        7abb97719928708e5aad850d6e5f7eff5dd5e5cd

---
 src/gui/AboutWidget.cpp    | 2 --
 src/gui/MainWidget.cpp     | 6 ++++++
 src/gui/SettingsWidget.cpp | 2 --
 3 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/gui/AboutWidget.cpp b/src/gui/AboutWidget.cpp
index f684c0a8..63d52e23 100644
--- a/src/gui/AboutWidget.cpp
+++ b/src/gui/AboutWidget.cpp
@@ -105,8 +105,6 @@ AboutWidget::AboutWidget()
     connect(closeB, SIGNAL(clicked()), this, SLOT(close()));
     connect(tabW, SIGNAL(currentChanged(int)), this, SLOT(currentTabChanged(int)));
     connect(&logWatcher, SIGNAL(fileChanged(const QString &)), this, SLOT(refreshLog()));
-
-    show();
 }

 void AboutWidget::showEvent(QShowEvent *)
diff --git a/src/gui/MainWidget.cpp b/src/gui/MainWidget.cpp
index b61316b7..92fa8fa1 100644
--- a/src/gui/MainWidget.cpp
+++ b/src/gui/MainWidget.cpp
@@ -1330,6 +1330,9 @@ void MainWidget::showSettings(const QString &moduleName)
             page = 0;

         settingsW = new SettingsWidget(page, moduleName, QMPlay2GUI.videoAdjustment);
+        if (windowFlags() & Qt::WindowStaysOnTopHint)
+            settingsW->setWindowFlag(Qt::WindowStaysOnTopHint);
+        settingsW->show();
         connect(settingsW, SIGNAL(settingsChanged(int, bool, bool)), &playC, SLOT(settingsChanged(int, bool, bool)));
         connect(settingsW, SIGNAL(setWheelStep(int)), seekS, SLOT(setWheelStep(int)));
         connect(settingsW, SIGNAL(setVolMax(int)), volW, SLOT(setMaximumVolume(int)));
@@ -1452,6 +1455,9 @@ void MainWidget::about()
     if (!aboutW)
     {
         aboutW = new AboutWidget;
+        if (windowFlags() & Qt::WindowStaysOnTopHint)
+            aboutW->setWindowFlag(Qt::WindowStaysOnTopHint);
+        aboutW->show();
         connect(aboutW, SIGNAL(destroyed()), this, SLOT(about()));
     }
     else
diff --git a/src/gui/SettingsWidget.cpp b/src/gui/SettingsWidget.cpp
index cdbba84c..410a7969 100644
--- a/src/gui/SettingsWidget.cpp
+++ b/src/gui/SettingsWidget.cpp
@@ -711,8 +711,6 @@ SettingsWidget::SettingsWidget(int page, const QString &moduleName, QWidget *vid

     connect(tabW, SIGNAL(currentChanged(int)), this, SLOT(tabCh(int)));
     tabW->setCurrentIndex(page);
-
-    show();
 }
 SettingsWidget::~SettingsWidget()
 {
